pkg/app/piped/platformprovider/lambda/client.go

```diff
@@ -295,6 +295,31 @@ func (c *client) updateFunctionConfiguration(ctx context.Context, fm FunctionMan
 	if !updateFunctionConfigurationSucceed {
 		return fmt.Errorf("failed to update configuration for Lambda function %s: %w", fm.Spec.Name, err)
 	}
+	if err := c.waitFunctionUpdated(ctx, fm.Spec.Name); err != nil {
+		return fmt.Errorf("failed to wait for Lambda function %s updated: %w", fm.Spec.Name, err)
+	}
+	return nil
+}
+
+func (c *client) waitFunctionUpdated(ctx context.Context, functionName string) error {
```

Comment:
How about moving this function body to L298? Looks like this function is specified to be used only by `updateFunctionConfiguration`, so I don't think we have enough reason to separate it as an independent function 🤔 Also, the current function name is quite common, `waitFunctionUpdated` while we have many `UpdateFunctionXXX`, which could cause misleading. So removing this function and moving all of its body to be part of `updateFunctionConfiguration` could be a better choice if you only want to wait after updating function configuration. wdyt?

Reply:
Okay.

> Also, the current function name is quite common, waitFunctionUpdated while we have many UpdateFunctionXXX, which could cause misleading.

I thinks so too.